gammapy/background/energy_offset_array.py

```diff
@@ -104,3 +127,73 @@ def plot_image(self, ax=None, offset=None, energy=None, **kwargs):
         image = ax.imshow(self.data, extent=extent, **kwargs)
         plt.colorbar(image)
         return ax
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyOffsetArray` from  fits file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdu_list = fits.open(filename)
+        return cls.from_fits(hdu_list)
+
+    @classmethod
+    def from_fits(cls, hdu):
+        """Read `EnergyOffsetArray` from a fits binary table.
+
+        Parameters
+        ----------
+        hdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the EnergyOffsetArray.
+
+        Returns
+        -------
+        cube : `~gammapy.background.EnergyOffsetArray`
+            EnergyOffsetArray object.
+        """
+
+        header = hdu[1].header
+        data = hdu[1].data
+        # get offset and energy binning
+        offset_edges = _make_bin_edges_array(data['THETA_LO'].squeeze(), data['THETA_HI'].squeeze())
+        energy_edges = _make_bin_edges_array(data['ENERG_LO'].squeeze(), data['ENERG_HI'].squeeze())
```

Comment:
Yes this is what I did for the moment I assumed TeV for energy and deg for offset!
Is the PR seems fine for you now?
Le 11 févr. 2016 à 14:37, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/background/energy_offset_array.py:
> 
> > +        Parameters
> > +        ----------
> > +        hdu : `~astropy.io.fits.BinTableHDU`
> > +            HDU binary table for the EnergyOffsetArray.
> > +
> > +        Returns
> > +        -------
> > +        cube : `~gammapy.background.EnergyOffsetArray`
> > +            EnergyOffsetArray object.
> > +        """
> > +
> > +        header = hdu[1].header
> > +        data = hdu[1].data
> > +        # get offset and energy binning
> > +        offset_edges = _make_bin_edges_array(data['THETA_LO'].squeeze(), data['THETA_HI'].squeeze())
> > +        energy_edges = _make_bin_edges_array(data['ENERG_LO'].squeeze(), data['ENERG_HI'].squeeze())
> The whole unit handling on FITS I/O needs to be separated out and re-used for all such classes.
> 
> Do whatever works for now ... you can even assume fixed units listed here:
> http://gamma-astro-data-formats.readthedocs.org/en/latest/irfs/background/index.html#bkg-2d-format
> (this is what Gammalib does at the moment ... they also need to implement unit handling on FITS I/O)
> 
> OK?
> 
> —
> Reply to this email directly or view it on GitHub.
>